main.go

```diff
@@ -150,8 +173,14 @@ func mailHandler(peer smtpd.Peer, env smtpd.Envelope) error {
 		peerIP = addr.IP.String()
 	}
 
-	log.Printf("new mail from=<%s> to=%s peer=[%s]\n", env.Sender,
-		env.Recipients, peerIP)
+	logger := log.WithFields(logrus.Fields{
+		"from": env.Sender,
+		"to": env.Recipients,
+		"peer": peerIP,
+		"host": *remoteHost,
+		"uuid": generateUUID(),
```

Comment:
I am really curious what this is good for. Should this look like an queue id from a store and forward mta to have an identifier for a single email? But then what is it good for if we only have it in one log message?

Reply:
This is not how we implemented it.

We used this UUID to correlate mail delivery attempts, successes and failures; having it in just one log entry will not be of much use.

@JonathonReinhart was this just an oversight or was it a conscious decision?

Reply:
This was carried over from #13 (with some enhancements). Note that this code is actually adding the `uuid` field to a local `logger` variable which is used for all subsequent log messages in this function. I presume the grafana guys did this to correlate log messages from before/after the sending takes place (which could take a while, and other logs could be interspersed).

Reply:
Ah, I missed that @JonathonReinhart - thanks. Yes, we're using the UUID to correlate the logs messages exactly as you describe there.

Reply:
I think @dannykopping is good with this, I'll let @decke resolve this conversation.

Reply:
Very nice feature!